fe/fe-core/src/main/java/com/starrocks/catalog/Table.java

```diff
@@ -742,4 +745,19 @@ public void setForeignKeyConstraints(List<ForeignKeyConstraint> foreignKeyConstr
     public List<ForeignKeyConstraint> getForeignKeyConstraints() {
         return this.foreignKeyConstraints;
     }
+
+    public synchronized List<Long> allocatePartitionIdByKey(List<PartitionKey> keys) {
+        long size = partitionKeyToId.size();
+        List<Long> ret = new ArrayList<>();
+        for (PartitionKey key : keys) {
+            Long v = partitionKeyToId.get(key);
+            if (v == null) {
+                partitionKeyToId.put(key, size);
```

Comment:
if partitionKeyToId reocrd entery （key1, 0）, but it will return list ret with value [1] ?

Reply:
No. why?